app/controllers/boards_controller.rb

```diff
@@ -106,7 +106,8 @@ def set_board
   end
 
   def boards_by_role(role)
-    Board.joins(:memberships).where(memberships: { user_id: current_user.id, role: role })
+    Board.where.not(id: Board.distinct.pluck(:previous_board_id))
```

Comment:
```ruby
Board.where.not(id: Board.select(:previous_board_id).where.not(previous_board_id: nil))

#=> Board Load (1.5ms)  SELECT "boards".* FROM "boards" WHERE "boards"."id" NOT IN (SELECT "boards"."previous_board_id" FROM "boards" WHERE "boards"."previous_board_id" IS NOT NULL) LIMIT $1  [["LIMIT", 11]]
```

That way it will not create extra request and make ruby array, which then will copy and paste, but will use subquery which is much faster.